app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
Ah, I thought the goal was to destroy the servers and keep the database in a legal state.

The second loop may raise an exception when the DecommissionProxy job tries to look up its proxy by id. Also, the third loop cannot be put in the Source base class, only Fog.